### ISSUE TITLE
Finetuning question

### ISSUE THREAD

**Issue by adbmdp** [xinyu1205/recognize-anything#173]
Hello,

I would like to finetune RAM++ tagging with other datasets.
I spent a lot of time trying to understand how it works.
But there are still quite a few points that I don't understand :-(

But before i ask my question, would it be possible, for example, to train the model on a dataset with personalities (pictures and name)?
So that RAM++ can tag them when I call inference_ram_plus.py?
A basic example would be a photo of Michael Jordan throwing a ball and i get a list of tags like:
`Micheal Jordan | basketball | basket | sport ...`

Also would it be possible to train the model with more complexe action pictures?
For example MMA / UFC (Mixed Martial Arts) pictures.
Then by analyzing an image, RAM++ would be able to give me a list of tags like:
`ground fight | strike | top mount position`
`stand up | uppercut`

Thanks!

**Comment by xinyu1205** [comment]
Thanks for your attention.
Actually, this is certainly feasible. The performance of the model depends on the quality of your finetune dataset.

**Comment by adbmdp** [comment]
Thanks for your reply and your awesome work @xinyu1205 !!

OK let's say I want to train the model with a celebrity dataset.

I have trouble understanding which tag file I need to update with the new tags.
To my understandings:
`parse_label_id` refers to the tag indices present in `ram/data/tag_list.txt`
`union_label_id` refers to the tag indices present in `ram/data/ram_tag_list.txt`

But for example when I watch in the COCO dataset for example can be found:

```
{
  "image_path":"coco/val2014/COCO_val2014_000000522418.jpg",
  "parse_label_id":[
    [
      4480,
      4532,
      678
    ]
  ],
  "caption":[
    "there is a woman that is cutting a white cake"
  ],
  "union_label_id":[
    4480,
    2624,
    2051,
    678,
    2599,
    2577,
    4532,
    1238,
    215,
    2332,
    4439
  ]
}
```
`parse_label_id` I should find the id in the `ram/data/tag_list.txt` file right?
This file only has 3429 IDs and I see an id 4480 !

So to summarize. If I want to modify only the tagging part of RAM++. In which file should I add my tags (maybe just one)?
And my Dataset can be something like:

```
{
  "image_path":"datasets/celebrities/CELEB_00001.jpg",
  "parse_label_id":[
    [
      9999
    ]
  ],
  "caption":[
    "Michael Jordan"
  ],
  "union_label_id":[
     8888
  ]
}
```

**Comment by xinyu1205** [comment]
parse_label_id refers to the tag parsed from image caption
union_label_id refers to the full tags of the image
Therefore, if you only have image-tag dataset, you just need set image tags as union_label_id.
And you only need the loss_tag and loss_dis in RAM or RAM++

**Comment by adbmdp** [comment]
Ok so I just need:
```
{
  "image_path":"datasets/celebrities/CELEB_00001.jpg",
  "caption":[
    "Michael Jordan"
  ],
  "union_label_id":[
     new id from ram/data/ram_tag_list.txt
  ]
}
```

> And you only need the loss_tag and loss_dis in RAM or RAM++
I don't know what you mean here but i'll try to find out. Do I have to change some code in finetune.py?

Thanks again for taking from you time to reply 👍 🥇

**Comment by xinyu1205** [comment]
It means you need to modify the forward function of ram.py or ram_plus.py.
And I strongly recommend that you read the RAM or RAM++paper before completing these tasks.

**Comment by adbmdp** [comment]
Thanks. I'll do that.

**Comment by aidyai** [comment]
Hi @adbmdp, where you able to fix these error?

**Comment by EricIrving-chs** [comment]
> d: cop

你好，我明白您的意思，并且知道如何解决。源代码在model.load_state_dict的时候会导入原有label_embed的模型部分，但事实上，在微调的时候我们会重新初始化一个（new_label_num, dim）的部分，新标签和原标签数量不一致，将会导致这个错误。你只需要在导入模型的时候删除label_embed部分即可。/ram/models/utils.py 在第275行加入代码：elif "label_embed" in k:
            del state_dict[k]
